src/test/java/com/google/api/codegen/testdata/go_mock_library.baseline

```diff
@@ -277,13 +285,492 @@ type mockLabeler struct {
     resps []interface{}
 }
 
-var _ taggerpb.LabelerServer = &mockLabeler{}
-
-func (s *mockLabeler) AddLabel(_ context.Context, req *taggerpb.AddLabelRequest) (*taggerpb.AddLabelResponse, error) {
+func (s *mockLabelerServer) AddLabel(_ context.Context, req *taggerpb.AddLabelRequest) (*taggerpb.AddLabelResponse, error) {
     s.reqs = append(s.reqs, req)
     if s.err != nil {
         return nil, s.err
     }
     return s.resps[0].(*taggerpb.AddLabelResponse), nil
 }
 
+
+// clientOpt is the option tests should use to connect to the test server.
+// It is initialized by TestMain.
+var clientOpt option.ClientOption
+
+var (
+    mockLibrary mockLibraryServer
+    mockLabeler mockLabelerServer
+)
+
+func TestMain(m *testing.M) {
+    flag.Parse()
+
+    serv := grpc.NewServer()
+    librarypb.RegisterLibraryServiceServer(serv, &mockLibrary)
+    taggerpb.RegisterLabelerServer(serv, &mockLabeler)
+
+    lis, err := net.Listen("tcp", "localhost:0")
+    if err != nil {
+        log.Fatal(err)
+    }
+    go serv.Serve(lis)
+
+    conn, err := grpc.Dial(lis.Addr().String(), grpc.WithInsecure())
+    if err != nil {
+        log.Fatal(err)
+    }
+    clientOpt = option.WithGRPCConn(conn)
+
+    os.Exit(m.Run())
+}
+
+func TestLibraryServiceCreateShelfError(t *testing.T) {
+    expErr := fmt.Errorf("test err %d", rand.Int())
+    mockLibrary.err = expErr
+
+    c, err := NewClient(context.Background(), clientOpt)
+    if err != nil {
+        t.Fatal(err)
+    }
+
+    var req *librarypb.CreateShelfRequest
+
+    reflect.ValueOf(&req).Elem().Set(reflect.New(reflect.TypeOf(req).Elem()))
+
+    _, err = c.CreateShelf(context.Background(), req)
+
+    if desc := grpc.ErrorDesc(err); desc != expErr.Error() {
```

Comment:
The message should be just for humans. Code should be able to change the message without breaking tests.